website_compassion/controllers/my_account.py

```diff
@@ -55,14 +56,38 @@ def _get_user_children(state=None):
     partner = request.env.user.partner_id
     only_correspondent = partner.app_displayed_sponsorships == "correspondent"
 
+    limit_date = datetime.now() - relativedelta(months=2)
+
+    exit_comm_config = list(
+        map(lambda x: partner.env.ref("partner_communication_switzerland." + x).id, ["lifecycle_child_planned_exit",
+                                                                                     "lifecycle_child_unplanned_exit"]))
+    exit_comm_to_send = partner.env["partner.communication.job"].search([
+        ("partner_id", "=", partner.id),
+        ("config_id", "in", exit_comm_config),
+        ("state", "=", "pending"),
+    ])
```

Comment:
The Child Exit communication will only be pending for a few moment (the time that the staff is working on it).
So in order to test if it's not already sent, it's better to check the inverse, check if one was already sent. You can also directly check if it relates to the sponsorship directly in the search.
```suggestion
    exit_comm_to_send = not partner.env["partner.communication.job"].search_count([
        ("partner_id", "=", partner.id),
        ("config_id", "in", exit_comm_config),
        ("state", "=", "done"),
        ("object_ids", "like", sponsorship.id)
    ])
```